Source/Model/FeatureConfig/AppLock/AppLockController.swift

```diff
@@ -122,18 +135,20 @@ public final class AppLockController: AppLockType {
     // Creates a new LAContext and evaluates the authentication settings of the user.
     public func evaluateAuthentication(scenario: AuthenticationScenario,
                                        description: String,
-                                       with callback: @escaping (AuthenticationResult, LAContext) -> Void) {
+                                       context: LAContextProtocol = LAContext(),
+                                       with callback: @escaping (AuthenticationResult, LAContextProtocol) -> Void) {
         
-        guard self.weakLAContext == nil else { return }
+//        guard self.weakLAContext == nil else { return }
         
-        let context: LAContext = LAContext()
         var error: NSError?
         
-        self.weakLAContext = context
+//        self.weakLAContext = context
         
         let canEvaluatePolicy = context.canEvaluatePolicy(scenario.policy, error: &error)
+        let shouldUseCustomPasscodeAsFallback = biometricsState.biometricsChanged(in: context) && (scenario.supportedUserFallback == .customPasscode)
```

Comment:
IMO this is easier to read if we rename `shouldUseCustomPasscodeAsFallback` to `biometricsChanged `.

```suggestion
        let biometricsChanged = biometricsState.biometricsChanged(in: context) && (scenario.supportedUserFallback == .customPasscode)
```